docs/source/howto/run_codes.rst

```diff
@@ -357,24 +357,215 @@ After :ref:`setting up your computer <how-to:run-codes:computer>` and :ref:`sett
+=================================================
+
+There are numerous reasons why you may need to re-run calculations you’ve already done before.
+Maybe you run a great number of complex workflows in high-throughput that each may repeat the same calculation, or you may have to restart an entire workflow that failed somewhere half-way through.
+Since AiiDA stores the full provenance of each calculation, it can detect whether a calculation has been run before and, instead of running it again, simply reuse its outputs, thereby saving valuable computational resources.
+This is what we mean by **caching** in AiiDA.
+
+.. _how-to:run-codes:caching:enable:
+
+How to enable caching
+---------------------
+
+Caching is **not enabled by default**.
+In order to enable caching for your profile (here called ``aiida_profile``), place the following ``cache_config.yml`` file in your ``.aiida`` configuration folder:
+
+.. code-block:: yaml
+
+    aiida_profile:
+        default: True
+
+From this point onwards, when you launch a new calculation, AiiDA will compare its hash (depending both on the type of calculation and its inputs, see :ref:`how-to:run-codes:caching:hashing`) against other calculations already present in your database.
+If another calculation with the same hash is found, AiiDA will reuse its results without repeating the actual calculation.
+
+In order to ensure that the provenance graph with and without caching is the same, AiiDA creates both a new calculation node and a copy of the output data nodes as shown in :numref:`fig_caching`.
+
+.. _fig_caching:
+.. figure:: include/images/caching.png
+    :align: center
+    :height: 350px
+
+    When reusing the results of a calculation **C** for a new calculation **C'**, AiiDA simply makes a copy of the result nodes and links them up as usual.
+
+.. note::
+
+    AiiDA uses the *hashes* of the input nodes **D1** and **D2** when searching the calculation cache.
+    I.e. if the input of **C'** were new nodes **D1'** and **D2'** with the same content (hash) as **D1**, **D2**, the cache would trigger as well.
+
+
+.. note:: Caching is **not** implemented at the WorkChain/workfunction level (see :ref:`how-to:run-codes:caching:limitations` for details).
+
+
+.. _how-to:run-codes:caching:hashing:
+
+How are nodes hashed
+--------------------
+
+*Hashing* is turned on by default, i.e. all nodes in AiiDA are hashed.
+This means that even when you enable caching once you have already completed a number of calculations, those calculations can still be used retro-actively by the caching mechanism since their hashes have been computed.
+
+The hash of a ``Data`` node is computed from:
+
+* all attributes of the node, except the ``_updatable_attributes`` and ``_hash_ignored_attributes``
+* the ``__version__`` of the package which defined the node class
+* the content of the repository folder of the node
+* the UUID of the computer, if the node is associated with one
```

Comment:
Since this is part of the How-Tos, I'm not sure that this level of detail concerning how the hash is computed is necessary. The important piece of information is the fact that the hash is unique to the node, no?

Reply:
I agree. I can move this to a Topics section and just link it here. Would that work?

Reply:
Yes, I think that would be ideal.